index.bs

```diff
@@ -2775,45 +4881,120 @@ partial interface MLGraphBuilder {
     tensor is specified by the *newShape* argument.
 </div>
 
-### The sigmoid() method ### {#api-mlgraphbuilder-sigmoid}
+<details open>
+  <summary>
+    The {{MLGraphBuilder/reshape(input, newShape)}} steps are:
+  </summary>
+  <div algorithm=reshape class=algorithm-steps>
+    1. If |input| is not an instance of {{MLOperand}}, then throw a "{{TypeError}}" {{DOMException}} and stop.
+    1. Let |outputShape| be an empty array of {{unsigned long}}.
+    1. If |newShape| is a scalar [=number=], set |outputShape| to `[ 1 ]`.
```

Comment:
> If |newShape| is a scalar [=number=]

Current the Spec only defines `newShape`  as **a sequence of {{nullable}} {{unsigned long}}**, how about creating a sperate PR of `|newShape| is a scalar [=number=]`, then Spec editors and more reviewers can easily review it and give feedbacks.

Reply:
Right, we should discuss this in an issue, not in this PR. Let's leave it for now and we can make changes after this PR.

Reply:
> Right, we should discuss this in an issue, not in this PR.

Yes, please open an issue for it, thanks.

Reply:
> If any value in newShape is 0, then [throw](https://webidl.spec.whatwg.org/#dfn-throw) a "[DataError](https://webidl.spec.whatwg.org/#dataerror)" [DOMException](https://webidl.spec.whatwg.org/#idl-DOMException).

Empty tensors are valid in ML. If the `input` tensor shape contains zeros, then it's correct for `newShape` to contain zeros. Indeed, if the `input` shape contains zeros, then `newShape` *must* have at least one zero, because otherwise that would be an error. It's only an error if the number of total input element count differs from the total output element count, and so it's good to remove this redundant clause because the element count check is both necessary and sufficient.

| oldShape | newShape | validity |
|------------|------------|----------|
| [1,2,3]       | [1,2,3]       | ✅        |
| [2,3,1]       | [3,2,1]       | ✅        |
| [1,2,3]       | [6]             | ✅        |
| [0,0,0]       | [0,0,0]        | ✅        |
| [1,2,0]       | [0,0,0]        | ✅        |
| [1,2,0]       | [0,2,1]        | ✅        |
| [1,2,3]       | [1,0,3]        | ❌        |
| [1,0,3]       | [1,2,3]        | ❌        |